### DISCUSSION TITLE
nextjs 12 with middleware - Prod i18n routing "randomly" broken since cli 28.11.1 release

### DISCUSSION THREAD

**Discussion by RandomDude123** [vercel/vercel#9220] [Help] [answered]
Next.js 12.3.1

https://www.getpliant.com/en/ -> served as always
https://www.getpliant.com/fi/  -> comes back empty

We currently have 8 locales and two of them don't work if one types the home url with locale slug in browser.
If one goes first to a working locale and uses an on-page link to navigate to `fi` it works.
Running the website locally via `npm run build && npm run start` works. Preview environments seem also to work fully.

First more locales were impacted - but after enough redeploys of the same branch (& commit) we currently "only" have 2 locales not working. We didn't find any recent changes made by us to the vercel configs, middleware or next config that could explain this behaviour.

In the middleware function logs on vercel are no errors logged and even the page requests which are not properly served are logged.

We opened a customer support case and waiting - but of course we need those locale homepages working asap.
Any ideas how this issue could be fixed OR at least what could be the root are much appreciated.

Is it possible to revert the vercel-cli version used for our prod deployments on vercel to check if it is the root cause?

**Comment by amyegan** [comment, marked as answer]
You can try an older version of the CLI by adding a `VERCEL_CLI_VERSION` environment variable with the value set to an older CLI version, such as `vercel@28.10.2`. Then redeploy or kick off a new deployment to use the specified version for the new build. If that works, please let me know so I can pull in the CLI team to help debug 🙏

**Reply by RandomDude123** [reply]
setting `VERCEL_CLI_VERSION  vercel@28.10.3` as env variable and deploy solved the issue. So i guess there could be something wrong in the last CLI version?

**Reply by amyegan** [reply]
Do you have a specific version of Next 12 pinned? The team suggested trying with the latest version if you have not already.

**Reply by RandomDude123** [reply]
we indeed had the nextjs version `12.3.1` pinned, after updating to `12.3.4` the vercel-cli version `28.11.1` (instead of `28.10.3`) works as expected. maybe there could be some info notification in the vercel project if the vercel-cli version is not set by an env variable (`VERCEL_CLI_VERSION`) and changed "under the hood". Took me quite some time to realise that the cli version changed...
